src/main/java/net/imglib2/type/PrimitiveTypeInfo.java

```diff
@@ -0,0 +1,98 @@
+package net.imglib2.type;
+
+import java.util.function.Function;
+
+import net.imglib2.img.NativeImg;
+import net.imglib2.img.basictypeaccess.ByteAccess;
+import net.imglib2.img.basictypeaccess.CharAccess;
+import net.imglib2.img.basictypeaccess.DoubleAccess;
+import net.imglib2.img.basictypeaccess.FloatAccess;
+import net.imglib2.img.basictypeaccess.IntAccess;
+import net.imglib2.img.basictypeaccess.LongAccess;
+import net.imglib2.img.basictypeaccess.ShortAccess;
+
+/**
+ * Binds a {@link PrimitiveType} enum constant to a {@code Access} interface
+ * ({@code ByteAccess}, {@code DoubleAccess}, and so on). Instances can only be
+ * constructed via static methods {@link #BYTE(Function)},
+ * {@link #DOUBLE(Function)}, etc. to prevent non-matching combinations of
+ * {@code Access} interface and {@link PrimitiveType} constant.
+ * <p>
+ * The second purpose of {@link PrimitiveTypeInfo} is to
+ * {@link #createLinkedType(NativeImg) create} a linked type {@code T} for a
+ * matching {@link NativeImg}.
+ *
+ * @param <T>
+ *            the {@link NativeType} this is attached to
+ * @param <A>
+ *            the {@code Access} family ({@code ByteAccess},
+ *            {@code DoubleAccess}, and so on)
+ *
+ * @author Tobias Pietzsch
+ */
+public final class PrimitiveTypeInfo< T extends NativeType< T >, A >
+{
```

Comment:
1. `PrimitiveTypeInfo` is again the wrong name. A better name would be `NativeTypeInfo` (every `NativeType` has a field of this type)
2. I would describe the purpose of this class differently:
    - The method `createLinkedType` returns a factory method to create a `NativeType`
    - The method `primitiveType` tell us which argument the factory method expects.

Conclusion: This class, represents a factory for a native type and provides some meta data, when it can be applied.
`NativeTypeFactory` would be a good name for this class.